### PR TITLE
Force re-read of commanded messages after control writes

### PR DESCRIPTION
**Pull request by tomquist**
## Summary

Settings changed from Home Assistant (e.g., *Discharge Depth*, *Charging Mode*) sometimes kept showing their old value for up to a minute. This fix ensures that the message containing a command is re-read immediately after the write, so the entity reflects what the device actually accepted instead of waiting up to a full polling interval.

## Changes

- **MqttClient**: Added `forceMessageIndices` option to `requestDeviceData()` to allow re-reading specific messages even when their poll interval has not elapsed. Forced messages still respect the `enabled` flag and `shouldPoll` predicate, and re-anchor the poll schedule to prevent a forced read from being immediately followed by a regular poll.

- **ControlHandler**: Modified the `publishCallback` signature to include a `messageIndex` parameter, identifying which message definition declares the command. This allows the caller to force re-read only that specific message after the write.

- **index.ts**: Updated the control handler callback to pass the message index and call `requestDeviceData()` with `forceMessageIndices` containing that index, ensuring the read-back happens within ~500ms instead of waiting up to a full polling interval.

- **Tests**: Added comprehensive test suite for forced refresh behavior, including edge cases (disabled messages, poll predicates, out-of-range indices). Updated existing test helpers to accommodate the new callback signature.

## Validation

- All existing tests pass
- New test suite covers forced refresh logic, schedule re-anchoring, and predicate/enabled checks
- No configuration changes required

https://claude.ai/code/session_01Sncjq1yPU8KPxFpfWNzG5D

## Summary by CodeRabbit

* **Bug Fixes**
  * Home Assistant settings now refresh within about a second instead of potentially remaining stale for up to a minute.
  * Device data refreshes immediately after control commands, ensuring updated values are read back reliably.
  * Forced refreshes respect enabled-state and polling eligibility rules without triggering unnecessary scheduled polls.
  * Disabled or ineligible data points are excluded from refresh requests.

### CODE REVIEW

File: src/mqttClient.ts [hunk 1]
```diff
@@ -360,7 +378,11 @@ export class MqttClient {
       }
       let lastRequestTimeKey = `${device.deviceId}:${idx}`;
       const lastRequestTime = this.lastRequestTime.get(lastRequestTimeKey);
-      if (lastRequestTime == null || now > lastRequestTime + message.pollInterval) {
+      if (
+        forced.has(idx) ||
+        lastRequestTime == null ||
+        now > lastRequestTime + message.pollInterval
+      ) {
```
[the remark this answers is not in the document]
Comment (tomquist) [reply src/mqttClient.ts:385]: Fixed in 8c7e63a — the preflight loop now skips disabled messages, with a regression test asserting no timeout is armed for an `enabled: false` message that has `controlsDeviceAvailability: true`.

Two notes on scope and severity:

No shipped device definition combines `enabled: false` with `controlsDeviceAvailability: true` — every message that can be disabled (`POLL_CELL_DATA`, `POLL_CALIBRATION_DATA`, `POLL_EXTRA_BATTERY_DATA`) sets `controlsDeviceAvailability: false`, and every availability-controlling message is unconditionally enabled. So this was latent, not a live false-offline. It's worth fixing anyway: a disabled message's `lastRequestTime` is never set, so it would read as due on *every* poll, and such a definition would take a device offline almost immediately.

The fix also made the per-index filtering in the forced set redundant. Both loops now skip disabled and predicate-excluded messages before consulting it, so the guarantee comes from the loops rather than being duplicated, and the set collapses to a plain `Set`. The three tests covering disabled / `shouldPoll: false` / out-of-range indices still pass and now exercise the loop-level behaviour.

The nitpick on the re-anchoring test was also correct, and worse than described — the test passed whether or not a forced read updated the timestamp. Fixed in the same commit: it now advances to t=60500, past the original deadline but short of the re-anchored one, and uses a non-availability message so the response timeout can't publish `offline` into the payload assertion. I verified both tests fail without their respective fixes.

---
_Generated by [Claude Code](https://claude.ai/code)_
